### ISSUE TITLE
Extra strings in module definition are ignored

### ISSUE THREAD

**Issue by dougbarth-stripe** [hashicorp/terraform#18374]
Terraform allows any number of strings after the module name, which are silently ignored.

### Terraform Version

```
Terraform v0.11.7
```

### Terraform Configuration Files

```hcl
# main.tf
module "foo" "bar" "baz" "boop" {
  source = "foo_module"
}

output "foo_thing" {
  value = "${module.foo.thing}"
}
```

```hcl
# foo_module/main.tf
output "thing" {
  value = "thing"
}
```

### Debug Output

https://gist.github.com/dougbarth-stripe/07032db84b9999780a9ce0ef75f82a9a

### Expected Behavior

Expected the parser to throw an error when running `terraform apply`

### Actual Behavior

The extra strings are ignored

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

**Comment by apparentlymart** [comment]
Hi @dougbarth-stripe! Sorry for this confusing behavior, and thanks for reporting it.

I believe this has already been fixed in our development branch for [the next major release](https://www.hashicorp.com/blog/terraform-0-1-2-preview), since the new configuration parser requires that the number of labels for each block type be defined as part of the configuration schema:

![Only 1 labels (name) are expected for module blocks.](https://user-images.githubusercontent.com/20180/42192410-9536f2e8-7e1d-11e8-8cf3-c6d099c7ec6e.png)

The awkward grammar of that error message aside, this was caught successfully by my build from the development branch. We'll use this issue as a reminder to verify the fix once the feature branch has been merged into `master`.

Thanks again for reporting this.

**Comment by apparentlymart** [comment]
Hi again, @dougbarth-stripe!

We now have that fix I mentioned merged in master, and so I re-verified with the v0.12.0-alpha2 prerelease build and got the same correct result.

We're going to just live with this weirdly-grammared message for now just for prioritization purposes but we'll return and clean that up in a later release; cleaning up error messages will be an ongoing effort just to due how many different ones there are in Terraform.

Since this is fixed in master, I'm going to close this out. The fix is ready to be included in the forthcoming v0.12.0 release. Thanks again for reporting this!
